test/index/functions/functions.snapshot.cc

```diff
@@ -0,0 +1,115 @@
+  
+  void top_level_func() {}
+//     ^^^^^^^^^^^^^^ definition [..] top_level_func(49f6e7a06ebc5aa8).
+  
+  namespace my_namespace {
+//          ^^^^^^^^^^^^ definition [..] my_namespace/
+    void func_in_namespace() {}
+//       ^^^^^^^^^^^^^^^^^ definition [..] my_namespace/func_in_namespace(49f6e7a06ebc5aa8).
+  }
+  
+  void overloaded_func(int) {}
+//     ^^^^^^^^^^^^^^^ definition [..] overloaded_func(d4f767463ce0a6b3).
+  void overloaded_func(const char *) {}
+//     ^^^^^^^^^^^^^^^ definition [..] overloaded_func(85c52e162fed56f9).
+  
+  void shadowed_func() {}
+//     ^^^^^^^^^^^^^ definition [..] shadowed_func(49f6e7a06ebc5aa8).
+  
+  namespace detail {
+//          ^^^^^^ definition [..] detail/
+    void shadowed_func() {}
+//       ^^^^^^^^^^^^^ definition [..] detail/shadowed_func(49f6e7a06ebc5aa8).
+  }
+  
+  using IntAlias = int;
+  void int_to_void_fn(int) {}
+//     ^^^^^^^^^^^^^^ definition [..] int_to_void_fn(d4f767463ce0a6b3).
+  void fn_with_same_canonical_type_and_hash_as_previous(IntAlias) {}
+//     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ definition [..] fn_with_same_canonical_type_and_hash_as_previous(d4f767463ce0a6b3).
```

Comment:
TODO: Maybe rename this to `same_hash_as_previous` to make the test more readable and add a comment that this is checking that the same canonical type produces the same hash.